rewrite-core/src/main/java/org/openrewrite/internal/ListUtils.java

```diff
@@ -355,6 +358,7 @@ public static <T> List<T> concat(@Nullable List<T> ls, @Nullable T t) {
      * @return A new list with the added element at the start, the original list if the element is null or a null
      * object if both element and list are null.
      */
+    @Contract("null, null -> null; !null, _ -> !null; _, !null -> !null")
     public static <T> @Nullable List<T> concat(@Nullable T t, @Nullable List<T> ls) {
```

Comment:
We could consider going one step further and also annotated the generic in the list being nullable:

```suggestion
    public static <T> @Nullable List<T> concat(@Nullable T t, @Nullable List<@Nullable T> ls) {
```

Reply:
Not sure. We don't actually filter out any nulls and we typically don't want to have nulls in our lists.

Reply:
Agreed, let's keep it this way. Now I think about it, you are right. We don't want nulls in our lists, so let's keep it tight (so non null!).